flink-ai-flow/ai_flow/api/ops.py

```diff
@@ -553,151 +426,159 @@ def user_define_operation(
-                               namespace: Text = DEFAULT_NAMESPACE
-                               ) -> None:
+    action_on_event(job_name=job_name,
+                    event_key=model_name,
+                    event_value="*",
+                    event_type=model_version_event_type,
+                    action=action,
+                    life=EventLife.ONCE,
+                    value_condition=MetValueCondition.UPDATE,
+                    condition=MetCondition.SUFFICIENT,
+                    namespace=namespace,
+                    sender=ANY_CONDITION)
+
+
+def action_on_dataset_event(job_name: Text,
+                            dataset_name: Text,
+                            action: TaskAction = TaskAction.START,
+                            namespace: Text = DEFAULT_NAMESPACE
+                            ) -> None:
     """
     Add dataset control dependency. It means src channel will start when and only the an new dataset of the specific
     dataset is updated in notification service.
-
     :param namespace: the namespace of the dataset
-    :param src: The src channel depended on the dataset which is updated in notification service.
+    :param job_name: The job name
     :param dataset_name: Name of the dataset, refers to a specific dataset.
-    :param dependency: The channel which is the dependency.
+    :param action:
     :return: None.
     """
-    tmp = DatasetControlEdge(dataset_name=dataset_name,
-                             target_node_id=dependency.node_id,
-                             source_node_id=src.node_id,
-                             namespace=namespace)
-    _default_ai_graph.add_edge(src.node_id, tmp)
-
-
-def user_define_control_dependency(src: Channel,
-                                   dependency: Channel,
-                                   event_key: Text,
-                                   event_value: Text,
-                                   event_type: Text = UNDEFINED_EVENT_TYPE,
-                                   condition: MetCondition = MetCondition.NECESSARY,
-                                   action: TaskAction = TaskAction.START,
-                                   life: EventLife = EventLife.ONCE,
-                                   value_condition: MetValueCondition = MetValueCondition.EQUAL,
-                                   namespace: Text = DEFAULT_NAMESPACE,
-                                   sender: Text = None
-                                   ) -> None:
+    action_on_event(job_name=job_name,
+                    event_key=dataset_name,
+                    event_value="created",
+                    event_type=AIFlowInnerEventType.DATASET_CHANGED,
+                    action=action,
+                    namespace=namespace,
+                    sender=ANY_CONDITION
+                    )
+
+
+def action_on_state(job_name: Text,
+                    upstream_job_name: Text,
+                    upstream_job_state: Text,
```

Comment:
Does the type of `upstream_job_state` need to use Enum? IMO, the users don't know how to set the value of this parameter if here doesn't use Enum.